### DISCUSSION TITLE
Migrating schema using Prisma, is it possible ?

### DISCUSSION THREAD

**Discussion by LilaRest** [directus/directus#18092] [Archive]
Hi !
I'm trying to use Prisma with Directus to create readable schema files, that can be easily pushed on Github, compared on PRs and then applied to production database.

Do you think that **introspecting** and creating **migrations files** using Prisma would involve any problem ?

Thanks in advance for your help :blush:

**Comment by rijkvanzanten** [comment]
That should work as expected. Just make sure to remove the `directus_` tables from prisma's tracking, as any migrations there might conflict with Directus' own migrations 👍🏻

**Reply by LilaRest** [reply]
Thanks for your quick answer !  

What if I want to extend one of the `directus_` tables locally and migrate it in prod ?

**Reply by rijkvanzanten** [reply]
I'm not too familiar with Prisma's setup, but just make sure you don't accidentally cause any conflicts with migrations that the system would introduce for those other core fields 👍🏻  As long as it's exclusively for a field that you _added_ to the system tables, it should be OK 🙂

**Reply by LilaRest** [reply]
Oh I got your point, it could lead to double migrations and unpredicable results. Is it possible to disable system migrations ?

**Reply by rijkvanzanten** [reply]
> Is it possible to disable system migrations ?

Noo, that's the way the system tables are installed / managed across upgrades. If you were to remove those, you wouldn't be able to install or update Directus itself!

**Reply by LilaRest** [reply]
> > Is it possible to disable system migrations ?
> 
> Noo, that's the way the system tables are installed / managed across upgrades. If you were to remove those, you wouldn't be able to install or update Directus itself!

Got it, have you an idea of how I can retrieve all the fields managed by the system or only the one added by the user on those tables ?

I plan to develop an extension that facilitates CI/CD between dev>staging>prod envs, are you working on a similar thing ? (don't want to duplicate)

**Reply by rijkvanzanten** [reply]
> Got it, have you an idea of how I can retrieve all the fields managed by the system or only the one added by the user on those tables ?

The system fields aren't tracked in directus_collections nor directus_fields, so that's the main source of truth of fields that we're added after the fact.

> I plan to develop an extension that facilitates CI/CD between dev>staging>prod envs, are you working on a similar thing ? (don't want to duplicate)

Uh depends! There's a lot of different ideas on what that would look like ideally (for example https://github.com/directus/directus/discussions/13041). As of right now, we don't have anything actively in the works yet, as we're still in that early R&D phase to figure out what the ideal "core" solution should look like 🙂

**Reply by LilaRest** [reply]
@rijkvanzanten Thanks for those details, I think I'll currently do without Directus until a "core" solution to that problem is released :)

Do you know if it's easily possible to use Directus as a content editor only ? 
I mean the DB, roles, permissions, users and all other contents would be logically handled by another service (e.g., Supabase) and Directus would just be used to edit all those data (from its awesome UI :)

And if yes, is it possible to install the Directus system tables in another DB so as not to clutter up the main one?

**Reply by rijkvanzanten** [reply]
> Do you know if it's easily possible to use Directus as a content editor only ?

Sure! It mirrors whatever's in the database, so if the content there is managed by something else at the same time, it'll be in sync in Directus as well 👍🏻 

> I mean the DB, roles, permissions, users and all other contents would be logically handled by another service (e.g., Supabase)

This is probably not going to work as you might expect. You still need a Directus User to login to the app/api, so you'd end up managing two different types of users in two different environments 👍🏻

**Reply by falmanna** [reply]
@LilaRest one way to utilize both supabase and directus is to keep directus for backoffice users (admins, devs, and content managers), while using supabase for app users.

Now, if you follow this approach then you will need to manage the app users' roles and permissions in supabase, while keeping directus roles and permissions management for the admin users. 
Not sure if you will need any extra shared logic, policies, or permissions as I didn't try this approach myself, but was thinking about it to get the best of both worlds. Although still happy with my directus only system.

**Reply by LilaRest** [reply]
> @LilaRest one way to utilize both supabase and directus is to keep directus for backoffice users (admins, devs, and content managers), while using supabase for app users.

That's an awesome idea, thanks for that !
I'll maybe give it a try in the next weeks :)

**Reply by jaymuk** [reply]
Did you ever try this @LilaRest ?

**Reply by LilaRest** [reply]
@jaymuk No I haven't actually tried that. I was looking for a more native / less tricky solution.
